megaqc/cli.py

```diff
@@ -25,17 +27,40 @@ def create_megaqc_app(info):
         CONFIG = ProdConfig()
     else:
         CONFIG = TestConfig()
+
+    if settings.run_db_check:
+        # Attempt to connect to the database exists to check that it exists
+        try:
+            dbengine = sqlalchemy.create_engine(
+                CONFIG.SQLALCHEMY_DATABASE_URI
+            ).connect()
+            metadata = sqlalchemy.MetaData(dbengine, reflect=True)
+            assert "sample_data" in metadata.tables
+        except:
+            print("\n##### ERROR! Could not find table 'sample_data' in database!")
+            print(
+                "Has the database been initialised? If not, please run 'megaqc initdb' first"
+            )
+            print("Exiting...\n")
+            sys.exit(1)
+        else:
+            dbengine.close()
+
     return create_app(CONFIG)
 
 
 @click.group(cls=FlaskGroup, create_app=create_megaqc_app)
-def cli():
+@click.pass_context
+def cli(ctx):
     """
     Welcome to the MegaQC command line interface.
 
     \nSee below for the available commands - for example,
     to start the MegaQC server, use the command: megaqc run
     """
+    # If the invoked command is not initdb we need to check whether a database already exists
+    if ctx.invoked_subcommand != "initdb":
+        settings.run_db_check = True
```

Comment:
Why do you need to set a setting here instead of simply running the check at this point?

Reply:
Depending on whether you call ```megaqc run``` or ```megaqc initdb``` different functions are evaluated first.
This is due to the FlaskGroup callback.

Try out the following:
Add a print statement into the `cli()` function and one into `create_megaqc_app`.
Then run both commands and see what happens. The check also needs to know the current config, which further makes this more annoying to deal with.

Reply:
Basically what I posted into the issue:

```
zeth@master ~/P/MegaQC (master)> megaqc initdb                                                                                                                                       (megaqc) 
This is MegaQC v0.3.0

 * Environment: test
 * Database type: sqlite
 * Database path: sqlite:////tmp/tmpuzxgmvav/megaqc.db
created
checked
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running stamp_revision  -> 4515f66f3690
Initialized the database.


zeth@master ~/P/MegaQC (master)> megaqc run                                                                                                                                          (megaqc) 
This is MegaQC v0.3.0

checked
 * Environment: test
 * Debug mode: off
 * Environment: test
 * Database type: sqlite
 * Database path: sqlite:////tmp/tmp71_b_ni4/megaqc.db
created
```